test/es6/default-splitscope.js

```diff
@@ -6,260 +6,622 @@
+               return b; 
+           } 
+           assert.areEqual(f2()(), 10, "Function defined in the param scope captures the formals from the param scope not body scope"); 
+
+           function f3(a = 10, b = function () { return a; }) { 
+               assert.areEqual(a, 1, "Initial value of parameter in the body scope should be the same as the one passed in"); 
+               var a = 20; 
+               assert.areEqual(a, 20, "Assignment in the body scope updates the variable's value in body scope"); 
+               return b; 
+           } 
+           assert.areEqual(f3(1)(), 1, "Function defined in the param scope captures the formals from the param scope even when the default expression is not applied for that param"); 
+
+           (function (a = 10, b = a += 10, c = function () { return a + b; }) { 
+               assert.areEqual(a, 20, "Initial value of parameter in the body scope should be same as the corresponding symbol's final value in the param scope"); 
+               var a2 = 40; 
+               (function () { assert.areEqual(a2, 40, "Symbols defined in the body scope should be unaffected by the duplicate formal symbols"); })(); 
+               assert.areEqual(c(), 40, "Function defined in param scope uses the formals from param scope even when executed inside the body"); 
+           })(); 
+
+           (function (a = 10, b = function () { assert.areEqual(a, 10, "Function defined in the param scope captures the formals from the param scope when executed from the param scope"); }, c = b()) { 
+           })(); 
+
+           function f4(a = 10, b = function () { return a; }) { 
+               a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f4()(), 10, "Even if the formals are not redeclared in the function body the symbol in the param scope and body scope are different"); 
+
+           function f5(a = 10, b = function () { return function () { return a; }; }) { 
+               var a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f5()()(), 10, "Parameter scope works fine with nested functions"); 
+
+           var a1 = 10; 
+           function f6(b = function () { return a1; }) { 
+               assert.areEqual(a1, undefined, "Inside the function body the assignment hasn't happened yet"); 
+               var a1 = 20; 
+               assert.areEqual(a1, 20, "Assignment to the symbol inside the function changes the value"); 
+               return b; 
+           } 
+           assert.areEqual(f6()(), 10, "Function in the param scope correctly binds to the outer variable"); 
+            
+           function f7(a = 10, b = { iFnc () { return a; } }) { 
+               a = 20; 
+               return b; 
+           } 
+           assert.areEqual(f7().iFnc(), 10, "Function definition inside the object literal should capture the formal from the param scope"); 
```

Comment:
expected and actual argument values are in the wrong order here and throughout this file.  They should be swapped.

Expected is first for readability.  And getting these in the wrong order makes the the error message when they fail incorrect, it'll have expected and actual reversed.